lib/router/router.js

```diff
@@ -15,6 +19,10 @@ export default class Router extends EventEmitter {
 
     // set up the component cache (by route keys)
     this.components = { [this.route]: { Component, err } }
+    // contain a map of response of prefetched routes
+    this.prefetchedRoutes = {}
+    this.prefetchingLockManager = new LockManager(2)
+    this.prefetchingRoutes = {}
```

Comment:
I wonder if we can simplify this state managements by having just Promise of `fetch` ?
Additionally, I think state of prefetch and normal fetch results can be managed by an object.